### PR TITLE
TK-454 add personalEvent and userEvent on the backend and frontend

### PR DESCRIPTION
**Pull request by Ven0mX**
- Update how to generate data for tests
- refactor code
- backend for userEvent
- frontend for userEvent 
- frontend for personnalEvent
- frontend for companyEvent

PS: BONNE CHANCE POUR LA REVIEW

### CODE REVIEW

File: src/main/java/fr/lunatech/timekeeper/services/UserEventService.java [hunk 1]
```diff
@@ -90,6 +114,21 @@ public Long createOrUpdateFromEventTemplate(EventTemplate eventTemplate,
         return updated;
     }
 
+    /**
+     * TODO add uservent rules
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/services/UserEventService.java]: Can you create a ticket to track this TODO on Jira please ? Thanks

File: src/main/java/fr/lunatech/timekeeper/services/UserEventService.java [hunk 2]
```diff
@@ -35,6 +37,28 @@
         return getEventsByUser(ownerId, monthNumber, year, TimeKeeperDateUtils::getMonthNumberFromDate);
     }
 
+    /**
+     * @param id
+     * @param context TODO SECURE ME WITH a CAN ACCESS
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/services/UserEventService.java]: Can you add a ticket on JIRA to track this task? Thanks

File: src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java [hunk 3]
```diff
@@ -51,15 +51,15 @@
     private final LocalDateTime endDateTime;
 
     @Null
-    private final List<EventTemplateResponse.Attendee> attendees;
+    private final List<User.Attendee> attendees;
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java]: For me, Attendee is some sort of DTO and thus, should be part of EventTemplateResponse. Not part of the model.

File: src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java [hunk 4]
```diff
@@ -124,55 +124,5 @@ public String toString() {
                 '}';
     }
 
-    public static final class Attendee {
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java:127]: I prefer to keep Attendee here, and not as part of the model, as it's a Data transfer object (and not something we really persist)

File: src/main/java/fr/lunatech/timekeeper/models/User.java [hunk 5]
```diff
@@ -59,7 +59,104 @@
...
+
+    public List<Profile> getProfiles() {
+        return profiles;
+    }
+
+    public List<ProjectUser> getProjects() {
+        return projects;
+    }
+
+    public static final class Attendee {
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/models/User.java]: I prefer to keep Attendee into the Response or Request object, and not as part of the model, as it's a Data transfer object (and not something we really persist)

File: src/main/java/fr/lunatech/timekeeper/services/responses/UserEventResponse.java [hunk 6]
```diff
@@ -36,6 +40,9 @@
     private String eventType;
     private String duration;
 
+    //wrap into a list in order to be maybe next feature, and it helps to be  with the front part
```
Comment (nicmarti) [inline src/main/java/fr/lunatech/timekeeper/services/responses/UserEventResponse.java]: which order ? 
create a ticket if you think it's important

File: src/test/java/fr/lunatech/timekeeper/resources/utils/DataTestProvider.java [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static fr.lunatech.timekeeper.services.responses.UserEventResponse.bind;
+
+@Singleton
+public class DataTestProvider {
```
Comment (nicmarti) [inline src/test/java/fr/lunatech/timekeeper/resources/utils/DataTestProvider.java:28]: That's a good idea 👍

File: src/test/java/fr/lunatech/timekeeper/resources/utils/ResourceDefinition.java [hunk 8]
```diff
@@ -36,8 +36,10 @@
     TimeSheetDef("/api/time-sheets", apply(TimeSheetRequest.class, TimeSheetResponse.class)),
     PersonalTimeSheetsWeekDef("/api/my/%d?weekNumber=%d", apply(Void.class, WeekResponse.class)),
     PersonalTimeSheetsMonthDef("/api/my/%d/month?monthNumber=%d", apply(Void.class, MonthResponse.class)),
-    EventDef("api/events", apply(EventTemplateRequest.class,EventTemplateResponse.class)),
-    EventUsersDef("api/events/%d/users", apply(Void.class,UserResponse.class));
+    PersonnalUserEventsDef("/api/user-events?userId=%d", apply(Void.class, UserEventResponse.class)),
+    UserEventsDef("/api/user-events", apply(UserEventRequest.class, UserEventResponse.class)),
+    TemplateEventDef("/api/events-template", apply(EventTemplateRequest.class, EventTemplateResponse.class)),
```
Comment (nicmarti) [inline src/test/java/fr/lunatech/timekeeper/resources/utils/ResourceDefinition.java:41]: good idea (events - template)

File: src/test/java/fr/lunatech/timekeeper/services/responses/UserEventResponseTest.java [hunk 9]
```diff
@@ -13,34 +14,43 @@
...
         var userEvent = new UserEvent();
+        var user = new User();
+        user.id = 1L;
         userEvent.id = 1L;
         userEvent.name = "Event name";
         userEvent.description = "Event description";
+        userEvent.owner = user;
         var response = UserEventResponse.createFromUserEvent(userEvent);
-
+        //TODO GUILLAUME CEST QUOI ÇA'??????????????????????????
```
Comment (nicmarti) [inline src/test/java/fr/lunatech/timekeeper/services/responses/UserEventResponseTest.java]: Please, no personal comment in the code (remember : the source code IS public and open-source)

**Review by nicmarti — CHANGES_REQUESTED** [review]

File: src/main/java/fr/lunatech/timekeeper/services/UserEventService.java (hunk 1 above)
Comment (Ven0mX) [reply src/main/java/fr/lunatech/timekeeper/services/UserEventService.java]: TK-473

File: src/main/java/fr/lunatech/timekeeper/services/UserEventService.java (hunk 2 above)
Comment (Ven0mX) [reply src/main/java/fr/lunatech/timekeeper/services/UserEventService.java]: TK-472

File: src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java (hunk 3 above)
Comment (Ven0mX) [reply src/main/java/fr/lunatech/timekeeper/services/responses/EventTemplateResponse.java]: yes of course, I move it into Response package

**Review by nicmarti — DISMISSED** [review]

**Comment by ani2fun** [comment]
Getting error when trying to access the http://localhost:3000/events

Try to switch various tabs among [User Events, Personal Events and Company Events] and you get this errors

Screenshot 2020-09-23 at 13 07 20

File: src/main/java/fr/lunatech/timekeeper/resources/openapi/UserEventResourceApi.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+                    description = "Invalid userId value"),
+            @APIResponse(
+                    responseCode = "403",
+                    description = "Invalid JWT token"),
+            @APIResponse(
+                    responseCode = "404",
+                    description = "user not found"
+            )
+    })
+    List<UserEventResponse> getPersonnalEvents(@QueryParam("userId") Long id);
```
Comment (ani2fun) [inline src/main/java/fr/lunatech/timekeeper/resources/openapi/UserEventResourceApi.java]: probably mis spelled here ... pls check getPersonalEvents or getPersonnelEvents

File: src/main/java/fr/lunatech/timekeeper/resources/EventTemplateResource.java [hunk 11]
```diff
@@ -51,12 +51,12 @@ public EventTemplateResponse getEventById(Long id) {
...
     }
 
-    @RolesAllowed({"user", "admin"})
+    @RolesAllowed({"admin"})
     @Override
     @Counted(name = "countGetAllEvents", description = "Counts how many times the user load the event list on method 'getAllEvents'")
     @Timed(name = "timeGetAllEvents", description = "Times how long it takes the user load the event list on method 'getAllEvents'", unit = MetricUnits.MILLISECONDS)
     public List<EventTemplateResponse> getAllEvents() {
-        return eventTemplateService.listAll(authentication.context());
+        return eventTemplateService.listEventCompany(authentication.context());
```
Comment (ani2fun) [inline src/main/java/fr/lunatech/timekeeper/resources/EventTemplateResource.java]: may be  `getAllCompanyEvents` ???

File: src/main/java/fr/lunatech/timekeeper/resources/PersonalResource.java [hunk 12]
```diff
@@ -51,7 +51,7 @@
     @Counted(name = "countGetPersonalWeek", description = "Counts how many times the user load his personal week on method 'getWeek'")
     @Timed(name = "timeGetPersonalWeek", description = "Times how long it takes the user load his personal week on method 'getWeek'", unit = MetricUnits.MILLISECONDS)
     public WeekResponse getWeek(Integer year, Integer weekNumber) {
-        if(logger.isDebugEnabled()) {
+        if (logger.isDebugEnabled()) {
```
Comment (ani2fun) [inline src/main/java/fr/lunatech/timekeeper/resources/PersonalResource.java:54]: just a question, do we really need logger.isDebugEnabled(), in Quarkus is not possible to change log levels in configuration file ...?

File: src/main/java/fr/lunatech/timekeeper/resources/UserEventResource.java [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+                .orElseThrow(NotFoundException::new);
+    }
+
+
+    @RolesAllowed({"user", "admin"})
+    @Override
+    @Counted(name = "countgetPersonnalEvents", description = "Counts how many times the user load the event list on method 'getPersonnalEvents'")
+    @Timed(name = "timegetPersonnalEvents", description = "Times how long it takes the user load the event list on method 'getPersonnalEvents'", unit = MetricUnits.MILLISECONDS)
+    public List<UserEventResponse> getPersonnalEvents(Long userId) {
+        return userEventService.listPersonnalEventForAnUser(userId, authentication.context());
```
Comment (ani2fun) [inline src/main/java/fr/lunatech/timekeeper/resources/UserEventResource.java]: may be `getAllPersonalEventsForAnUser`

File: src/main/java/fr/lunatech/timekeeper/resources/PersonalResource.java (hunk 12 above)
Comment (Ven0mX) [reply src/main/java/fr/lunatech/timekeeper/resources/PersonalResource.java:54]: that's not my code, so I have to investigate

**Review by ani2fun — APPROVED** [review]
LGTM

**Pull request merged**